community/kernel/src/test/java/org/neo4j/kernel/recovery/LatestCheckPointFinderTest.java

```diff
@@ -78,327 +79,415 @@ public LatestCheckPointFinderTest( Integer startLogVersion, Integer endLogVersio
-        setupLogFiles( startLogVersion, endLogVersion );
-        LatestCheckPointFinder finder = new LatestCheckPointFinder( logFiles, fs, reader );
-        LogEntryStart start = new LogEntryStart( 0, 0, 0, 0, new byte[0], new LogPosition( endLogVersion, 16 ) );
-        CheckPoint checkPoint = new CheckPoint( new LogPosition( endLogVersion, 33 ) );
-        when( reader.readLogEntry( any( ReadableLogChannel.class ) ) ).thenReturn( start, checkPoint, null );
+        setupLogFiles( logFile( start(), checkPoint() ) );
 
         // when
         LatestCheckPoint latestCheckPoint = finder.find( endLogVersion );
 
         // then
-        assertEquals( new LatestCheckPoint( checkPoint, false, endLogVersion ), latestCheckPoint );
+        assertLatestCheckPoint( true, false, NO_TRANSACTION_ID, endLogVersion, latestCheckPoint );
     }
 
     @Test
     public void latestLogFileContainingACheckPointAndAStartAfter() throws Throwable
     {
         // given
-        setupLogFiles( startLogVersion, endLogVersion );
-        LatestCheckPointFinder finder = new LatestCheckPointFinder( logFiles, fs, reader );
-        CheckPoint checkPoint = new CheckPoint( new LogPosition( endLogVersion, 16 ) );
-        LogEntryStart start = new LogEntryStart( 0, 0, 0, 0, new byte[0], new LogPosition( endLogVersion, 33 ) );
-        when( reader.readLogEntry( any( ReadableLogChannel.class ) ) ).thenReturn( start, checkPoint, null );
+        long txId = 35;
+        StartEntry start = start();
+        setupLogFiles( logFile( start, commit( txId ), checkPoint( start ) ) );
 
         // when
         LatestCheckPoint latestCheckPoint = finder.find( endLogVersion );
 
         // then
-        assertEquals( new LatestCheckPoint( checkPoint, true, endLogVersion ), latestCheckPoint );
+        assertLatestCheckPoint( true, true, txId, endLogVersion, latestCheckPoint );
     }
 
     @Test
-    public void latestLogFileContainingACheckPointAndAStartAtSamePosition() throws Throwable
+    public void latestLogFileContainingACheckPointAndAStartWithoutCommitAfter() throws Throwable
     {
         // given
-        setupLogFiles( startLogVersion, endLogVersion );
-        LatestCheckPointFinder finder = new LatestCheckPointFinder( logFiles, fs, reader );
-        CheckPoint checkPoint = new CheckPoint( new LogPosition( endLogVersion, 16 ) );
-        LogEntryStart start = new LogEntryStart( 0, 0, 0, 0, new byte[0], new LogPosition( endLogVersion, 16 ) );
-        when( reader.readLogEntry( any( ReadableLogChannel.class ) ) ).thenReturn( start, checkPoint, null );
+        StartEntry start = start();
+        setupLogFiles( logFile( start, checkPoint( start ) ) );
 
         // when
         LatestCheckPoint latestCheckPoint = finder.find( endLogVersion );
 
         // then
-        assertEquals( new LatestCheckPoint( checkPoint, true, endLogVersion ), latestCheckPoint );
+        assertLatestCheckPoint( true, true, NO_TRANSACTION_ID, endLogVersion, latestCheckPoint );
```

Comment:
Is this test really correct?
There is no commit entry in the log but we still expect `commitsAfterLastCheckPoint` to be true.